cpp/arcticdb/column_store/column.hpp

```diff
@@ -528,6 +525,18 @@ class Column {
         return *data_.buffer().ptr_cast<T>(bytes_offset(*physical_row), sizeof(T));
     }
 
+    // Copies all physical scalars to a std::vector<T>. This is useful if you require many random access operations
+    // and you would like to avoid the overhead of computing the exact location every time.
+    template<typename T>
+    std::vector<T> clone_scalars_to_vector() const {
+        auto values = std::vector<T>(row_count());
```

Comment:
Maybe calling reserve + push_back could be faster

Reply:
I did the change. This is not part of the bottleneck so I don't expect benchmarks to change but still did it as it seems nicer.